packages/user/src/context/UserProvider.tsx

```diff
@@ -48,14 +48,21 @@ const UserProvider = ({ children }: Properties) => {
 
     if (user) {
       roles = await getUserRoles();
-
-      isEmailVerified = await isUserVerified();
-      await setUserData({
-        ...user,
-        roles: roles,
-        isEmailVerified: isEmailVerified,
-      });
-      setUser({ ...user, roles: roles, isEmailVerified: isEmailVerified });
+      if (appConfig.user.features?.signUp?.emailVerification) {
+        isEmailVerified = await isUserVerified();
+        await setUserData({
+          ...user,
+          roles: roles,
+          isEmailVerified: isEmailVerified,
+        });
+        setUser({ ...user, roles: roles, isEmailVerified: isEmailVerified });
+      } else {
+        await setUserData({
+          ...user,
+          roles: roles,
+        });
+        setUser({ ...user, roles: roles });
+      }
```

Comment:
I think we can simplify this code. Use the if else to change the user data and then finally do setUserData and setUser at the end.
```
let userData = {
  ...user,
  roles:
}

// if else here

// finally
await setUserData(userData)
setUser(userData);
```

Reply:
Also could we change the function name to `updateUser`? setVerifiedUserWithRoles is very long, not correctly meaningful and not extendable.